src/openrct2-ui/windows/ServerList.cpp

```diff
@@ -267,22 +268,17 @@ class ServerListWindow final : public Window
         if (text.empty())
             return;
 
-        std::string temp = static_cast<std::string>(text);
+        u8string temp = static_cast<u8string>(text);
 
         switch (widgetIndex)
         {
             case WIDX_PLAYER_NAME_INPUT:
                 if (_playerName == text)
                     return;
 
-                text.copy(_playerName, sizeof(_playerName));
-
-                // Don't allow empty player names
-                if (_playerName[0] != '\0')
-                {
-                    gConfigNetwork.PlayerName = _playerName;
-                    ConfigSaveDefault();
-                }
+                _playerName = temp;
+                gConfigNetwork.PlayerName = _playerName;
+                ConfigSaveDefault();
```

Comment:
More of a remark than something to change right now. This gets called with every character you type. I think it would be nicer to only save it when the input box loses focus, or when Enter is pressed or something.

Reply:
I agree, I gave it a shot here so it saves on window close or loss of focus: [c6e2546](https://github.com/OpenRCT2/OpenRCT2/pull/20004/commits/c6e2546accada9b558422bde983a43943cdd5d84)

Reply:
Nice to see you picked this up. I left a few notes on the implementation.